src/zenml/materializers/built_in_materializer.py

```diff
@@ -75,6 +80,23 @@ def save(self, data: Any) -> None:
         super().save(data)
         yaml_utils.write_json(self.data_path, data)
 
+    def extract_metadata(
+        self, data: Union[bool, float, int, str]
+    ) -> Dict[str, "MetadataType"]:
+        """Extract metadata from the given built-in container object.
+
+        Args:
+            data: The built-in container object to extract metadata from.
+
+        Returns:
+            The extracted metadata as a dictionary.
+        """
+        base_metadata = super().extract_metadata(data)
+        builtin_metadata = {
+            "string_representation": str(data),
```

Comment:
Yeah the idea was if you have a basic data type it's probably most useful to just show the value. 

For `int`/`float`/`bool` this is always reasonably small. For `str` this could become very big though, I even encountered that already with the evidently materializer that returns the entire dashboard visualization as a 5M char `str`. That's why I included a manual check somewhere to filter such metadata out.

Reply:
maybe we could be more restrictive, numbers and booleans seem safe

Reply:
Yeah I changed this to only include numbers and booleans